src/methods.ts

```diff
@@ -8,19 +8,46 @@ import {
     SendTransactionRequest,
     TransactionHash,
     Empty,
+    InvokeContractRequest,
 } from '../grpc/concordium_p2p_rpc_pb';
 import NodeClient from './client';
 import { invalidParameterError, nodeError } from './errors';
 import {
+    isHex,
     isValidAccountAddress,
     isValidBase64,
     isValidCredentialId,
     isValidHash,
     isValidUInt64,
+    isValidContractAddress,
     validateParams,
 } from './validation';
 import JSONbig from 'json-bigint';
 
+type ContractAddress = {
+    index: number | bigint;
+    subindex: number | bigint;
+};
+type Invoker =
+    | {
+          type: 'AddressContract';
+          address: ContractAddress;
+      }
+    | {
+          type: 'AddressAccount';
+          address: string;
+      }
+    | null;
+
+interface ContractContext {
```

Comment:
Are these types not available from the sdk?

Reply:
The ContractAddress and ContractContext are not the same as the ones in the SDK. Invoker is (currently) only a local type.

Reply:
Okay. But please move the types to a separate file.